docs/modules/httpd.md

```diff
@@ -0,0 +1,286 @@
+  return {
+    status = "200 OK",
+    type = "text/plain",
+    getbody = function()
+      local data = f:read(512) -- pick a suiteable chunk size here
+      if not data then f:close() end
+      return data
+    end,
+  }
+end
+
+httpd.dynamic(httpd.GET, "/mysensordata", handler)
+```
+
+## httpd.start()
+Starts the web server. The server has to be started before routes can be
+configured.
+
+#### Syntax
+```lua
+httpd.start({
+  webroot = "<static file prefix>",
+  max_handlers = 20,
+  auto_index = httpd.INDEX_NONE || httpd.INDEX_ROOT || httpd.INDEX_ALL,
+})
+```
+
+#### Parameters
+A single configuration table is provided, with the following possible fields:
+
+  - `webroot` (mandatory) This sets the prefix used when serving static files.
+  For example, with `webroot` set to "web", a HTTP request for "/index.html"
+  will result in the httpd module trying to serve the file "web/index.html"
+  from the file system. Do NOT set this to the empty string, as that would
+  provide remote access to your entire virtual file system, including special
+  files such as virtual device files (e.g. "/dev/uart1") which would likely
+  present a serious security issue.
+  - `max_handlers` (optional) Configures the maximum number of route handlers
+  the server will support. Default value is 20, which includes both the
+  standard static file extension handlers and any user-provided handlers.
+  Raising this will result in a bit of additional memory being used. Adjust
+  if and when necessary.
+  - `auto_index` Sets the indexer mode to be used. Most web servers
+  automatically go looking for an "index.html" file when a directory is
+  requested. For example, when pointing your web browser to a web site
+  for the first time, e.g. http://www.example.com/ the actual request will
+  come through for "/", which in turn commonly gets translated to "/index.html"
+  on the server. This behaviour can be enabled in this module as well. There
+  are three modes provided:
+    - `httpd.INDEX_NONE` No automatic translation to "index.html" is provided.
+    - `httpd.INDEX_ROOT` Only the root ("/") is translated to "/index.html".
+    - `httpd.INDEX_ALL` Any path ending with a "/" has "index.html" appended.
+    For example, a request for "subdir/" would become "subdir/index.html",
+    which in turn might result in the file "web/subdir/index.html" being
+    served (if the `webroot` was set to "web").
+    The default value is `httpd.INDEX_ROOT`.
+
+#### Returns
+`nil`
+
+#### Example
+```lua
+httpd.start({ webroot = "web", auto_index = httpd.INDEX_ALL })
+```
+
+## httpd.stop()
+
+Stops the web server. All registered route handlers are removed.
+
+#### Syntax
+```lua
+httpd.stop()
+```
+
+#### Parameters
+None.
+
+#### Returns
+`nil`
+
+
+## httpd.static()
+
+Registers a static route handler.
+
+#### Syntax
+```
+httpd.static(route, content_type)
+```
+
+#### Parameters
+- `route` The route prefix. Typically in the form of \*.ext to serve all files
```

Comment:
I don't think that you want the backslash here

Reply:
Don't I? Without that it'll start an italic section, will it not?

Reply:
The joy and wonders of loosely defined Markdown 😄 Using a backslash means being on the safe side here as we have to cater for at least two interpreters: GitHub (GFM) and MkDocs. Did you consider putting *.ext in "" in both places?

Reply:
Dare I ask what having it double-quoted would do to the markdown parsers?

Reply:
I didn't mention this wrt the parser, sorry. It's just that once it'll read 'the ".ext" extension' and once 'the form of *.ext to serve' (w/o quotes here).

Reply:
Ah, right. So `*.ext` is the wildcard form needed to register a handler for all files with the extension `.ext`. Maybe I should switch to code format and drop the quotes?